llm_benchmark.py

```diff
@@ -261,28 +261,60 @@ async def openai_embed(context: ApiContext) -> ApiResult:
     return await post(context, url, headers, data)
 
 
+def make_anthropic_messages(prompt: str, files: Optional[List[InputFile]] = None):
+    """Formats the prompt as a text chunk and any images as image chunks.
+    Note that Anthropic's image protocol is somewhat different from OpenAI's."""
+    if not files:
+        return [{"role": "user", "content": prompt}]
+
+    content: List[Dict[str, Any]] = [{"type": "text", "text": prompt}]
+    for file in files:
+        if not file.mime_type.startswith("image/"):
+            raise ValueError(f"Unsupported file type: {file}")
+        source = {
+            "type": "base64",
+            "media_type": file.mime_type,
+            "data": file.base64_data,
+        }
+        content.append({"type": "image", "source": source})
+    return [{"role": "user", "content": content}]
+
+
 async def anthropic_chat(context: ApiContext) -> ApiResult:
+    """Make an Anthropic chat completion request. The request protocol is similar to OpenAI's,
+    but the response protocol is completely different."""
+
     async def chunk_gen(response) -> TokenGenerator:
+        tokens = 0
         async for chunk in make_sse_chunk_gen(response):
-            yield chunk.get("completion", "")
-
-    url = "https://api.anthropic.com/v1/complete"
+            delta = chunk.get("delta")
+            if delta and delta.get("type") == "text_delta":
+                tokens += 1
+                yield delta["text"]
+            usage = chunk.get("usage")
+            if usage:
+                num_tokens = usage.get("output_tokens")
+                while tokens < num_tokens:

```

Comment:
Is this worth logging? Seems like it artificially increases TPS

Reply:
It turns out that not every token results in a chunk, so we emit some fake chunks at the end to get to the right number. Ultimately we should just emit this number directly, but the work needed for this seemed like it was going to be more than I wanted to take on in this PR.